core/src/main/java/org/elasticsearch/common/util/primitives/Integers.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.util.primitives;
+
+import com.google.common.primitives.Ints;
+import org.elasticsearch.common.Strings;
+
+import java.util.*;
+
+public class Integers {
+    /**
+     * Tries to parse the given String to an int
+     *
+     * @param value the String to try to parse to an int
+     * @return the parsed value as an int or null if the String can not be parsed to an int
+     */
+    public static Integer tryParse(String value) {
+        if (Strings.isNullOrEmpty(value)) {
```

Comment:
@rmuir The leniency is needed because of the `state` directory that sits along side the shard subdirectories. Perhaps a good solution to this problem is to just ignore that one directory and just straight up parse (no leniency) the rest of the subdirectories so remove the need for this lenient method?